src/components/pairSelector.tsx

```diff
@@ -15,13 +15,13 @@ export default function PairSelector({
   setShow: (show: boolean) => void;
   onSelect: (assets: [Token, Token]) => void;
 }) {
+  const tokenAssetsList = useTokenAssets();
   const [tokenAssets, setTokenAssets] = useState<Record<string, Token>>({});
   const [selectedAssets, setSelectedAssets] = useState<Token[]>([]);
 
   useEffect(() => {
-    const tokenAssets = fetchAllTokenAssets();
-    setTokenAssets(Object.fromEntries(tokenAssets.map(asset => [asset.symbol, asset])));
-  }, []);
+    setTokenAssets(Object.fromEntries(tokenAssetsList.map(asset => [asset.symbol, asset])));
+  }, [tokenAssetsList]);
```

Comment:
This should be in react-query

Reply:
Correct, but let's keep that to another PR (also this is v1 code).